### ISSUE TITLE
no-loop-func warnings at ForInStatement.right and ForStatement.init.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#2427]
I got `no-loop-func` warnings at this code:

```js
"use strict";
let xs = [1, 2, 3];
for (let x of xs.filter(x0 => x0 != null)) {
  //                    ^^ here!
  console.log(x);
}
for (let x in xs.filter(x0 => x0 != null)) {
  //                    ^^ here!
  console.log(x);
}
for (let i = 0, ys = xs.filter(x0 => x0 != null); i < ys.length; i++) {
  //                           ^^ here!
  console.log(ys[i]);
}
```

But I think that `ForInStatement.right` and `ForStatement.init` are out of loop, since those run once at first.

I'm happy to make PR if this is accepted :)

----

version: `v0.20.0`
.eslintrc:

```
{
  "env": {
    "node": true,
    "es6": true
  },
  "ecmaFeatures": {
    "arrowFunctions": true,
    "forOf": true
  }
}
```

**Comment by nzakas** [comment]
Can you show the actual ESLint output?

**Comment by mysticatea** [comment]
Sorry, I had checked on Sublime Text.

```
test.js
   3:24  error  Don't make functions within a loop  no-loop-func
   7:24  error  Don't make functions within a loop  no-loop-func
  11:31  error  Don't make functions within a loop  no-loop-func

✖ 3 problems (3 errors, 0 warnings)
```

I expect there aren't errors.

**Comment by nzakas** [comment]
Yeah, that's a bug. We should omit the `init` portion from this check.
